bindings/cpp/newapi/bulk_remove_handler.cpp

```diff
@@ -0,0 +1,216 @@
+		                {"keys", m_keys.size()},
+		                {"trace_id", to_hex_string(m_session.get_trace_id())},
+		               });
+	}
+	if (m_keys.empty()) {
+		m_handler.complete(create_error(-ENXIO, "send_bulk_remove: keys list is empty"));
+		return;
+	}
+
+	// group keys
+	std::map<dnet_addr, std::vector<std::pair<dnet_id, dnet_time>>> remote_ids; // node_address -> [list of keys]
+	const bool has_direct_address = !!(m_session.get_cflags() & (DNET_FLAGS_DIRECT | DNET_FLAGS_DIRECT_BACKEND));
+
+	if (!has_direct_address) {
+		// prepare handler for error of getting address
+		auto error_handler = [&](const dnet_id& key, const int& err) {
+			dnet_addr address;
+			memset(&address, 0, sizeof(address));
+			dnet_cmd cmd;
+			memset(&cmd, 0, sizeof(cmd));
+			cmd.cmd = DNET_CMD_BULK_REMOVE_NEW;
+			cmd.trace_id = m_session.get_trace_id();
+			cmd.flags = DNET_FLAGS_REPLY | DNET_FLAGS_MORE;
+			if (m_session.get_trace_bit())
+				cmd.flags |= DNET_FLAGS_TRACE_BIT;
+
+			cmd.id = key;
+			cmd.status = err;
+			auto result_data = std::make_shared<callback_result_data>(&address, &cmd);
+			result_data->error = create_error(err,
+			                                  "bulk_remove_handler: could not locate address & "
+			                                  "backend for requested key: %s",
+			                                  dnet_dump_id(&key));
+			ioremap::elliptics::callback_result_entry entry(result_data);
+			process(callback_cast<remove_result_entry>(entry));
+		};
+
+		split_keys_to_nodes(m_session, m_keys, remote_ids, error_handler);
+	}
+	else {
+		const auto address = m_session.get_direct_address();
+		remote_ids.emplace(address.to_raw(), m_keys);
+	}
+	
+	std::vector<async_remove_result> results;
+	results.reserve(remote_ids.size());
+
+	for (const auto &pair : remote_ids) {
+		const dnet_addr &address = pair.first;
+		const dnet_bulk_remove_request request(pair.second);
+		const auto packet = serialize(request);
+
+		transport_control control;
+		control.set_command(DNET_CMD_BULK_REMOVE_NEW);
+		control.set_cflags(m_session.get_cflags() | DNET_FLAGS_NEED_ACK );
+		control.set_data(packet.data(), packet.size());
+
+		auto session = m_session.clean_clone();
+		if (!has_direct_address)
+			session.set_direct_id(address);
+
+		results.emplace_back(session); 
+		auto handler = std::make_shared<single_bulk_remove_handler>(results.back(), session, address);
+		handler->start(control, request);
+	}
+
+	auto rr = aggregated(m_session, results);
+	m_handler.set_total(rr.total());
+
+	rr.connect(
+		std::bind(&bulk_remove_handler::process, shared_from_this(), std::placeholders::_1),
+		std::bind(&bulk_remove_handler::complete, shared_from_this(), std::placeholders::_1)
+	);
+
+}
+
+void bulk_remove_handler::process(const remove_result_entry &entry) {
+	m_handler.process(entry);
+
+	const auto *cmd = entry.command();
+	m_transes.emplace(cmd->trans);
+	auto it = m_statuses.emplace(entry.status(), 1);
```

Comment:
You can write `++m_statuses[entry.status()];`.
`std::map` and `std::unordered_map` initialize new values by `T()`